### DISCUSSION TITLE
Facing problem while creating a tree from a nested recursive dictionary

### DISCUSSION THREAD

**Discussion by chaihub** [kayjan/bigtree#234] [Questions] [answered]
Greetings!

I am using Python 3.11 with bigtree version 0.17.2. I am trying to create a tree from my data that looks like the below dictionary:

```python
data = {
    "organization": {
        "id": "1234",
        "name": "Jack",
	"role": "Lead",
        "children": [
            {
                "id": "5678",
                "name": "Jack",
		"role": "Developer"
            }
        ]
    }
}
```

In the data, there is a 'name' key in each node but its values are not unique, hence I need to use 'id' as the key to identify the tree nodes.

When I use the below function to create the tree:

```python
org_root = nested_dict_to_tree(data["organization"], name_key="id")
```

the operation fails and I get the error messages:

```
  File "bigtree_test.py", line 70, in <module>
    org_root = nested_dict_to_tree(data["data"], name_key="id")
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>\Python\Python311\Lib\site-packages\bigtree\tree\construct.py", line 752, in nested_dict_to_tree
    root_node = _recursive_add_child(node_attrs)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>\Python\Python311\Lib\site-packages\bigtree\tree\construct.py", line 747, in _recursive_add_child
    node = node_type(node_name, parent=parent_node, **child_dict)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Node.__init__() got multiple values for argument 'name'
python-BaseException
```

This happens when both, 'id' and 'name' keys are present together. However, if I change 'name' to 'nam' or 'name2', there is no problem and the tree is created successfully.

Am I doing something wrong? Or is this a bug?

Thanks for your help!

**Comment by kayjan** [comment, marked as answer]
Hello, this is intended as `name` is a special keyword when creating `Node` object. This prevents having `name` as a keyword/identifier, and having `name` again as an attribute.

In your case, if the unique key is `id`, when performing `nested_dict_to_tree` operation, the name of the Node reads from `id` under the hood, which clashes with the `name` attribute that you have. Happy to consider how else to handle this better because of the special keyword constraint from `Node` itself.

**Reply by chaihub** [reply]
Thank you for your quick response. In an ideal scenario, it would be great to be able to use something other than `name` as the unique key and still also have `name` as another key. I suspect it should be possible to keep the `Node` attribute separate from the JSON key, but I am not an expert and am not sure about it.
